src/util/std_expr.h

```diff
@@ -4611,7 +4614,10 @@ class class_method_descriptor_exprt : public nullary_exprt
     return get(ID_component_name);
   }
 
-  const irep_idt &get_class_name() const
+  /// Unique identifier in the symbol table, of the compile time type of the
+  /// class which this expression is applied to. For example this could be -
+  /// `java::java.lang.Object`.
+  const irep_idt &class_id() const
```

Comment:
:question: not sure I agree with rename - it is the classes symbol name, I'd have thought the identifier as as without the java:: prefix... actually not sure I agree with myself here, seems we use tag / id(entifier) / name throughout. Perahps you have a convincing argument for why this is a good rename?

Reply:
To me an `_id` is a unique identifier, whereas a name might be a "friendlier" identifier and is not guaranteed to be unique. Because the return of this function is a unique identifier, I think the function should be named `_id`.

Reply:
IMHO `name` and `id` should both mean the same thing, i.e. unique if said unqualified. But `id` is a bit more common for this use so I think it’s good to leave this as is. For human readable representations we have the `pretty_` prefix elsewhere